src/pudl/extract/phmsagas.py

```diff
@@ -52,6 +52,32 @@ def process_renamed(self, newdata: pd.DataFrame, page: str, **partition):
                     f"\n{to_drop}"
                 )
                 newdata = newdata.drop(columns=to_drop, errors="ignore")
+        # there is an annoying middling number of columns in phmsa raw data that are unnamed
+        # and have a smattering of random values. we want to drop these guys. but we are going
+        # to enumerate what we expect to need to
+        unnamed_page_years = {
+            "yearly_distribution": [
+                2000,
+                2001,
+                2002,
+                2003,
+                2004,
+                2005,
+                2006,
+                2007,
+                2009,
+            ]
+        }
+        unnamed_columns = newdata.filter(like="unnamed").columns
+        if (page in unnamed_page_years) and (
+            int(partition["year"]) in unnamed_page_years[page]
+        ):
+            newdata = newdata.drop(columns=unnamed_columns)
+        elif not unnamed_columns.empty:
+            logger.warning(
+                "We found some unnamed columns that are probably not expected. "
+                f"Consider dropping them. Columns found: {unnamed_columns}"
+            )
```

Comment:
there were several instances of `unnamed_X` columns in PHMSA. I considered adding them all into the column maps but that would just mean definitely keeping track of trash columns. In 2009 there were a ton of extra columns seemingly from two records that had a multi-line comment that shifted the rest of the cells over. The options were to map a bunch of bad columns and maybe save a few records in a really bespoke way or just drop em all. this seemed like the easiest and most discrete way to drop em all rather than map these all in the column maps.